core/src/worker/slot_provider.rs

```diff
@@ -0,0 +1,199 @@
+//! This module implements traits defined in the client to dispatch a
+//! WFT to a worker bypassing the server.
+//! This enables latency optimizations such as Eager Workflow Start.
+
+use anyhow::Context;
+use std::fmt::{Debug, Formatter};
+
+use crate::{
+    abstractions::{MeteredSemaphore, OwnedMeteredSemPermit},
+    protosext::ValidPollWFTQResponse,
+    worker::workflow::wft_poller::validate_wft,
+};
+
+use std::sync::Arc;
+use temporal_client::{Slot as SlotTrait, SlotProvider as SlotProviderTrait};
+use temporal_sdk_core_protos::temporal::api::workflowservice::v1::PollWorkflowTaskQueueResponse;
+use tokio::sync::mpsc::UnboundedSender;
+use tonic::Status;
+
+type WFTStreamSender =
+    UnboundedSender<Result<(ValidPollWFTQResponse, OwnedMeteredSemPermit), Status>>;
+
+pub struct Slot {
+    permit: Option<OwnedMeteredSemPermit>,
```

Comment:
This doesn't need to be an option any more, once you make `schedule_wft` consume `self`

Reply:
Done